admin/src/main/scala/io/buoyant/admin/names/DelegateHandler.scala

```diff
@@ -1,21 +1,37 @@
 package io.buoyant.admin.names
 
 import com.twitter.finagle.http.{Request, Response}
+import com.twitter.finagle.naming.NameInterpreter
 import com.twitter.finagle.{Dtab, Namer, Path, Service}
 import com.twitter.util.Future
 import io.buoyant.admin.HtmlView
-import io.buoyant.linkerd.admin.names.DelegateApiHandler
+import io.buoyant.namer.DelegatingNameInterpreter
 
 class DelegateHandler(
   view: HtmlView,
-  dtabs: () => Future[Map[String, Dtab]],
-  namers: Seq[(Path, Namer)]
+  baseDtabs: Map[String, Dtab],
+  interpreter: String => NameInterpreter
 ) extends Service[Request, Response] {
 
-  def apply(req: Request): Future[Response] =
-    dtabs().map(render).flatMap(view.mkResponse(_))
+  def apply(req: Request): Future[Response] = {
+    Future.collect {
+      baseDtabs.map {
+        case (ns, dtab) =>
+          interpreter(ns) match {
+            case delegating: DelegatingNameInterpreter =>
+              delegating.dtab.values.toFuture.flatMap(Future.const)
+                .map { interpreterDtab =>
+                  ns -> interpreterDtab
+                }
+            case _ => Future.value(ns -> Dtab.empty)
+          }
+      }.toSeq
+    }.map(_.toMap).map { interpreterDtabs =>
+      render(interpreterDtabs, baseDtabs)
+    }.flatMap(view.mkResponse(_))
+  }
```

Comment:
I generally prefer to break up complex blocks into a few clear statements:

e.g.

```scala
def apply(req: Request): Future[Response] = {
  val dtabs = baseDtabs.map {
    case (ns, dtab) => getDtab(ns).map(ns -> _)
  }
  Future.collect(dtabs.toSeq)
    .map { ds => render(ds.toMap, baseDtabs) }
    .flatMap(view.mkResponse(_))
}
```

I sometimes find it hard to understand chains of collect/map/flatmap, and I find it helpful to see them split out.